hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/state/DispatchingEvmFrameState.java

```diff
@@ -91,26 +91,40 @@ public class DispatchingEvmFrameState implements EvmFrameState {
 
     private static final String ADDRESS_BYTECODE_PATTERN = "fefefefefefefefefefefefefefefefefefefefe";
 
+    private static final String PROXY_PRE_BYTES = "6080604052348015600f57600080fd5b506000610";
```

Comment:
I do like this version better.  I was actually thinking of something like this (two alternatives here, 1 and 2):

```java
    private static final String PROXY_PRE_BYTES = "6080604052348015600f57600080fd5b506000610";
    private static final String PROXY_MID_BYTES = "905077";
    private static final String PROXY_POST_BYTES =
            "fefefefefefefefefefefefefefefefefefefefe600052366000602037600080366018016008845af43d806000803e8160008114605857816000f35b816000fdfea2646970667358221220d8378feed472ba49a0005514ef7087017f707b45fb9bf56bb81bb93ff19a238b64736f6c634300080b0033";

    private static final BiFunction<String, String, String> constructProxy1 =
            (@NonNull final String systemContractAddress, @NonNull final String functionSelectorHex) ->
                    PROXY_PRE_BYTES + systemContractAddress + PROXY_MID_BYTES + functionSelectorHex + PROXY_POST_BYTES;

    private static final String PROXY_METHOD_TEMPLATE =
            "6080604052348015600f57600080fd5b506000610%1$s905077%2$sfefefefefefefefefefefefefefefefefefefefe600052366000602037600080366018016008845af43d806000803e8160008114605857816000f35b816000fdfea2646970667358221220d8378feed472ba49a0005514ef7087017f707b45fb9bf56bb81bb93ff19a238b64736f6c634300080b0033";
    private static final BiFunction<String, String, String> constructProxy2 =
            (@NonNull final String systemContractAddress, @NonNull final String functionSelectorHex) ->
                    PROXY_METHOD_TEMPLATE.formatted(systemContractAddress, functionSelectorHex);

    private static final String TOKEN_CALL_REDIRECT_CONTRACT_BINARY = constructProxy1.apply("0167", "618dc65e");
    private static final String ACCOUNT_CALL_REDIRECT_CONTRACT_BINARY = constructProxy2.apply("016a", "e4cbd3a7");
```